### PR TITLE
Create action to pull AppMaps from CircleCI and Analyze

### PR DESCRIPTION
**Pull request by petecheslock**

### CODE REVIEW

File: .github/workflows/analyze-maps-from-circleci.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                sleep 10
+              fi
+          done
+
+      - name: Download and extract appmaps
+        run: | 
+          aws s3 cp s3://${appmap_storage_bucket}/${base_sha}.tar
+          tar xvf ${base_sha}.tar
+
+      - name: Archive AppMaps
```
[the remark this answers is not in the document]
Comment (petecheslock) [reply .github/workflows/analyze-maps-from-circleci.yml:58]: Thanks - going to look and see if i can adjust this or check for the existence of an artifact already saved then maybe skip this if exists.

[the remark this answers is not in the document]
Comment (kgilpin) [reply .github/workflows/analyze-maps-from-circleci.yml:58]: You could try `appmap restore --exact` and then if that fails you know you need to build it.

[the remark this answers is not in the document]
Comment (petecheslock) [reply .github/workflows/analyze-maps-from-circleci.yml:58]: That's so funny I was just wondering while looking at the action code if that would be the way to do it.  I'm going to give that a shot, should work just like the other s3 copy commands.

File: .github/workflows/analyze-maps-from-circleci.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        with:
+          install-appmap-library: false
+
+      - name: Check for existing baseline archive
+        id: check_for_baseline
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        continue-on-error: true  
+        run: |
+          appmap restore --revision ${base_sha} --github-repo ${GITHUB_REPOSITORY} --exact
```
Comment (petecheslock) [inline .github/workflows/analyze-maps-from-circleci.yml:36]: While this is fine and works - if we had a `--check` flag or something to check the existence of the archive vs doing the full download that would avoid some unnecessary network.

Comment (petecheslock) [reply .github/workflows/analyze-maps-from-circleci.yml:36]: Cool - i'll take a look!  Thanks for the link

**Pull request closed without merging**